gtsfm/utils/prepare_nerfstudio.py

```diff
@@ -0,0 +1,159 @@
+"""Functions to create transforms.json file and resized images that are input to Nerfstudio
+
+Here I have modified code taken from Nerfstudio for parsing data in the Nerfstudio format.
+Original file at:
+https://github.com/nerfstudio-project/nerfstudio/blob/main/scripts/process_data.py.
+
+Author: Jon Womack
+"""
+import argparse
+import os
+import json
+import numpy as np
+from enum import Enum
+from pathlib import Path
+
+
+from gtsfm.utils import images, io
+import thirdparty.nerfstudio.colmap_utils as colmap_utils
+
+
+class CameraModel(Enum):
+    """Enum for camera types."""
+
+    OPENCV = "OPENCV"
+    OPENCV_FISHEYE = "OPENCV_FISHEYE"
+
+
+CAMERA_MODELS = {
+    "perspective": CameraModel.OPENCV,
+    "fisheye": CameraModel.OPENCV_FISHEYE,
+}
+
+
+def colmap_to_json(cameras_path: Path, images_path: Path, output_dir: Path, camera_model: CameraModel) -> int:
+    """Converts COLMAP's cameras.bin and images.bin to a JSON file.
+    Args:
+        cameras_path: Path to the cameras.txt file.
+        images_path: Path to the images.txt file.
+        output_dir: Path to the output directory.
+        camera_model: Camera model used.
+    Returns:
+        The number of registered images.
+    """
+
+    cameras = colmap_utils.read_cameras_text(cameras_path)
+    images = colmap_utils.read_images_text(images_path)
+
+    # Only supports one camera
```

Comment:
Does this mean all the cameras should share the same params?

Reply:
Yes, as Nerfstudio's parser does: https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/process_data/colmap_utils.py#L595